Configurations/windows-makefile.tmpl

```diff
@@ -773,13 +773,24 @@ EOF
               # If the target is named foo.S in build.info, we want to
               # end up generating foo.s in two steps.
               if ($args{src} =~ /\.S$/) {
-                   return <<"EOF";
+                if(($config{CC} !~ /clang-cl/) and ($config{AS} =~ /clang-cl/))
+                {
+                    return <<"EOF";
 $target: "$gen0" $deps
 	cmd /C "set "ASM=\$(AS)" & $generator \$@.S"
-	\$(CPP) $incs $cppflags $defs \$@.S > \$@.i
+    \$(CPP) /D__ASSEMBLER__ /E $incs $cppflags $defs \$@.S > \$@.i
```

Comment:
Drop `/E`, as it's already present (well, `/EP` is...).  Also, don't convert the initial TAB to spaces.